docs/pages/versions/unversioned/sdk/in-app-purchases.md

```diff
@@ -155,15 +170,21 @@ if (responseCode === ResponseCode.OK) {
 
 ### `InAppPurchases.purchaseItemAsync(productId: string, oldItem?: string)`
 
-Initiates the purchase flow to buy the item associated with this productId. This function is void and the result must be handled in the callback that you passed in to `setPurchaseListener`. This will display a prompt to the user which will allow them to either buy the item or cancel the purchase.
+Initiates the purchase flow to buy the item associated with this productId. This function returns null as the result must be handled in the callback that you passed in to `setPurchaseListener`. This will display a prompt to the user which will allow them to either buy the item or cancel the purchase.
```

Comment:
The TS source code returns `Promise<void>`, which is conceptually correct here. Since the body of the TS function just awaits the native method (rather than `return await`) -- the returned promise definitely resolves to undefined (`Promise<void>`)

Also, the promise's value doesn't really matter -- just the fact that this method is async and returns a promise to await is more important. So phrasing the docs to be focused more on awaiting the promise and the timing of when it's resolved helps highlight what's important here, ex:

"This function returns a void promise that resolves when <describe condition here>. 

---

Separately, I find this part confusing: 
> ...that you passed in to `setPurchaseListener`. This will display a prompt...

Does "This" refer to `purchaseItemAsync` or the purchase listener?

Reply:
Changed in latest commits to:
> Initiates the purchase flow to buy the item associated with this `productId`. This will display a prompt to the user that will allow them to either buy the item or cancel the purchase. When the purchase completes, the result must be handled in the callback that you passed in to `setPurchaseListener`.

The prompt is displayed as a result of calling `purchaseItemAsync`. The purchase listener doesn't receive the result until the user clicks "Buy" or "Cancel".